core/txpool/txpool.go

```diff
@@ -486,6 +486,7 @@ func (p *TxPool) Sync() error {
 
 // Clear removes all tracked txs from the subpools.
 func (p *TxPool) Clear() {
+	p.Sync()
```

Comment:
```suggestion
        // invoke Sync to ensure that txs pending addition don't get added to the pool after
        // the subpools are subsequently cleared
	p.Sync()
```

Reply:
We ignore the potential error from `Sync` but this is okay:  it only returns an error if the pool is closed, and calling `Clear` on a closed pool should be fine (TM).